aten/src/ATen/native/TensorFactories.h

```diff
@@ -64,5 +64,25 @@ inline void check_size_nonnegative(IntArrayRef size) {
     TORCH_CHECK(x >= 0, "Trying to create tensor with negative dimension ", x, ": ", size);
   }
 }
+
+inline void check_supported_max_int_with_precision(int64_t n, const Tensor& tensor) {
+  TORCH_CHECK(at::scalar_tensor(n, tensor.options()).defined(),
```

Comment:
I'm confused about the error checking here.  Do you know if it's actually possible to trigger this TORCH_CHECK?  I can't see how it is, but maybe I'm missing something.

Reply:
This was moved out from the original CUDA implementation. To me it seems checking for situation like the following:

    torch.randperm(100000, dtype=torch.half, device='cuda')                                                               

(half cannot represent a number as large as 100000.)

Practically it gives me a different error (before I made this change, so it would check for the cuda function only):
                               
    RuntimeError: value cannot be converted to type at::Half without overflow: 100000

It doesn't seem helpful, but IMO it won't hurt to stay either.

Reply:
sure, sounds reasonable.